Issue #1382: Added short and consistent map format names to use with --export-map

MarceColl opened the pull request:
For Issue #1382 
Added a new virtual method to MapFormat called `shortName` to use whenever a short descriptive name is needed. Particularly for CLI input/output.

Added short names for all map format plugins.

`--export-formats` now uses the `shortName()`
`--export-map` now uses the `shortName()`
`--export-map` with no filter argument still uses `nameFilter()`

I've used the names of the modules as reference, if any of them are not appropriate I will change them.

bjorn commented on src/plugins/python/pythonplugin.cpp:
```diff
@@ -363,6 +363,32 @@ QString PythonMapFormat::nameFilter() const
+    PyObject *pfun = PyObject_GetAttrString(mClass, "shortName");
+    if (!pfun || !PyCallable_Check(pfun)) {
+        PySys_WriteStderr("Plugin extension doesn't define \"shortName\"\n");
+        return ret;
```
Maybe it would be helpful to fall back to returning the result from `nameFilter` here, for backwards compatibility?

MarceColl commented:
Added requested changes.
Sorry, there were some tr() functions that I forgot. Changed them to QLatin1String now. Made all shortName methods public and the python shortName fallbacks to nameFilter when shortName doesn't exist. I'm not sure if we should keep the warning that shortName is not implemented tho.

bjorn commented:
> I'm not sure if we should keep the warning that shortName is not implemented tho. I think we should if shortName is the expected implementation. Something like: Plugin extension doesn't define shortName. Fallbacking to nameFilter?

I agree. Though please use "Falling back" instead of "Fallbacking". :-P

MarceColl commented:
Okay, done!

bjorn commented:
Awesome, thanks a lot!

MarceColl commented:
Thank you for Tiled! I will start working on the y-offset tiles this afternoon.

bjorn commented:
@MarceColl You're welcome! I'm looking forward to your next PR. :-)
